src/object/srv_mod.c

```diff
@@ -35,6 +37,7 @@ obj_mod_init(void)
 {
 	dss_abt_pool_choose_cb_register(DAOS_OBJ_MODULE,
 					ds_obj_abt_pool_choose_cb);
+	vos_dtx_register_check_leader(ds_pool_check_leader);
```

Comment:
minor, must to register a callback, cannot directly call the ds_pool_check_leader() instead?

Reply:
There is compile error if we call ds_pool_check_leader() from VOS directly.